### DISCUSSION TITLE
How can I do some actions before exit on tauri v2?

### DISCUSSION THREAD

**Discussion by tmg0** [tauri-apps/tauri#10531] [Q&A] [answered]
I wanna handle some actions when the app exits, such as terminating a process manually.

In [`Plugin Development`](https://v2.tauri.app/develop/plugins/#on_event) document, I found plugin `Builder` provide `on_event` to math `RunEvent`, but I cannot find some methods to do something like that in `tauri::Builder`. It seems only provide `on_window_event` and `on_webview_event`?

Besides using plugins, are there any other ways to access `lifecycle events`?

**Comment by FabianLars** [comment, marked as answer]
Replace `builder.run().expect()` with `builder.build().expect().run()` and then the `run` function will give you access to the RunEvents: https://github.com/tauri-apps/tauri/blob/dev/examples/api/src-tauri/src/lib.rs#L146-L152

**Comment by tmg0** [comment]
Thank you very much for your quick answer 🚀 It solved my problem!
